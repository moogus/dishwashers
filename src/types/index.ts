export type GridItemModel = {
  productId: string;
  price: string;
  title: string;
  image: string;
};

type AdditionalServices = {
  includedServices: string[];
}

export type ProductModel = {
  title: string;
  media: Images,
  price: string;
  productInformation: string;
  displaySpecialOffer: string;
  additionalServices: AdditionalServices;
  code: string;
}

type Price = {
  now?: string
}

export type ApiGridItemModel = {
  productId: string;
  price: Price;
  title: string;
  image: string;
}

type Images = {
  altText: string;
  urls: string[]
}

type Media = {
  images: Images;
}

type Details = {
  productInformation: string;
}

export type ApiProductModel = {
  title : string;
  media: Media;
  price?: Price;
  details?: Details;
  displaySpecialOffer : string;
  additionalServices: AdditionalServices;
  code : string;
}

export type ThunkConfig = {
  gridApi: string;
  productApi: string;
}