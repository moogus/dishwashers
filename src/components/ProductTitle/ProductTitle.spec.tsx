import React from "react";
import { render } from "@testing-library/react";
import { mocked } from "ts-jest/utils";

import useGetProduct from "../../hooks/product/get-product";
import { ProductModel } from "../../types";

import { ProductTitle } from "./index";

jest.mock("../../hooks/product/get-product");

describe("<ProductTitle/>", () => {
  const renderContainer = () => render(<ProductTitle />);

  describe("when <ProductTitle/> is rendered", () => {
    beforeEach(() => {
      mocked(useGetProduct).mockClear();
      mocked(useGetProduct).mockReturnValue({
        title: "what the dickens",
      } as ProductModel);
    });

    test("component renders as expected", () => {
      const { container } = renderContainer();
      expect(container).toMatchSnapshot();
    });
  });
});
