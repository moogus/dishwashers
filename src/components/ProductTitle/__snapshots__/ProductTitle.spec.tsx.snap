// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTitle/> when <ProductTitle/> is rendered component renders as expected 1`] = `
<div>
  <h1
    class="sc-bdfBwQ eChoRV pt-4 pb-4"
  >
    what the dickens
  </h1>
</div>
`;
